### ISSUE TITLE
GenerateDescriptorAsync doesn't use the correct header property 

### ISSUE THREAD

**Issue by daniel-pebble** [oras-project/oras-dotnet#133]
### What happened in your code base?

When attempting to use FetchAsync to download a manifest it fails as Oras isn't looking at the correct location for the "Docker-Content-Digest" header.

I've adjusted the code to check both response.Content.Headers and response.Headers and in my tests its always on response.Headers.

### What did you expect to happen?

Content digest is found in the correct header location

### How can we reproduce it?

```
if (response.Content.Headers.TryGetValues("Docker-Content-Digest", out var serverContentHeaderDigest))
{
    serverDigest = serverContentHeaderDigest.FirstOrDefault();
    if (!string.IsNullOrEmpty(serverDigest))
    {
        response.VerifyContentDigest(serverDigest);
    }
}

if (response.Headers.TryGetValues("Docker-Content-Digest", out var serverHeaderDigest))
{
    serverDigest = serverHeaderDigest.FirstOrDefault();
    if (!string.IsNullOrEmpty(serverDigest))
    {
        response.VerifyContentDigest(serverDigest);
    }
}
```

### What is the version or commit of the ORAS .NET library?

Commit: c58adc96ac2d31cbd98d4c2e5211981326151ee4

### What are your OS and Runtime environments?

Windows 11 .net 8

### Are you willing to submit PRs to fix it?

- [X] Yes, I am willing to fix it.

**Comment by Wwwsylvia** [comment]
Hi @daniel-pebble, thanks for the issue! 

> I've adjusted the code to check both response.Content.Headers and response.Headers and in my tests its always on response.Headers.

I just took a look at the [Azure .NET SDK](https://github.com/Azure/azure-sdk-for-net/blob/3ef52b7b1b5cb30b77f39bc9484ec0b5719ae10b/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs#L641) and found that `response.Headers` is checked there.

```csharp
rawResponse.Headers.TryGetValue("Docker-Content-Digest", out string responseHeaderDigest);
```

Would you like to go ahead and create a PR to fix this?

**Comment by shizhMSFT** [comment]
Closed by #136
